src/test/java/io/airlift/slice/TestSlices.java

```diff
@@ -159,5 +160,8 @@ public void testEnsureSize()
         slice = ensureSize(fourBytes, 5);
         assertEquals(slice.length(), 8);
         assertEquals(slice.getBytes(), new byte[] {1, 2, 3, 4, 0, 0, 0, 0});
+
+        slice = ensureSize(allocate(Integer.MAX_VALUE - 100), Integer.MAX_VALUE - 50);
```

Comment:
Maybe use `Integer.MAX_VALUE / SLICE_ALLOW_SKEW + 100` instead of `allocate(Integer.MAX_VALUE - 100)`. Maybe it will pass then (but probably not)

Reply:
I went further and used `Integer.MAX_VALUE / SLICE_ALLOW_SKEW^37` instead. This is just above SLICE_ALLOC_THRESHOLD, so the value computed by `ensureSize` will remain close to `Integer.MAX_VALUE`

Reply:
I went further and used `(Integer.MAX_VALUE+100) / SLICE_ALLOW_SKEW^17` instead. This is ~48M (so above SLICE_ALLOC_THRESHOLD) so the value computed by `ensureSize` will remain close to `Integer.MAX_VALUE+100` + capping on double-to-int cast, so `Integer.MAX_VALUE`.